tests/Unit/ContentTypeSnapshotTest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * This file is part of the contentful-management.php package.
+ *
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Tests\Unit;
+
+use Contentful\Management\ResourceBuilder;
+
+class ContentTypeSnapshotTest extends \PHPUnit_Framework_TestCase
+{
+    public function testJsonSerialize()
+    {
+        $builder = new ResourceBuilder();
+
+        $data = [
+            'sys' => [
+                'type' => 'Snapshot',
+                'snapshotEntityType' => 'ContentType',
+            ],
+            'snapshot' => [
+                'name' => 'Versioned Content Type',
+                'displayField' => 'title',
+                'fields' => [
+                    [
+                        'name' => 'Title',
+                        'id' => 'title',
+                        'type' => 'Symbol',
+                    ],
+                    [
+                        'name' => 'Description',
+                        'id' => 'description',
+                        'type' => 'Text',
+                    ],
+                ],
+                'sys' => [
+                    'id' => 'versionedContentType',
+                    'type' => 'ContentType',
+                ],
+            ],
+        ];
+
+        $entrySnapshot = $builder->buildObjectsFromRawData($data);
```

Comment:
We already had this discussion (I think it was when implementing webhooks) and you said yourself you didn't have a better idea 😅